mcs_benchmark_data/cli/template_contexts/metadata_template_context.py

```diff
@@ -0,0 +1,34 @@
+from typing import Optional
+from dataclasses import dataclass
+from pathlib import Path
+
+from mcs_benchmark_data.cli.template_contexts.file_context import FileContext
+
+
+@dataclass(init=True)
+class MetadataTemplateContext(FileContext):
+    """
+    Information about the metadata file necessary to create a file from a template
+    """
+
+    def __init__(self, benchmark_name, submission_name=None):
+        self.benchmark_name = benchmark_name
+        self.submission_name = submission_name
+        self.template_name = (

```

Comment:
Ah, I see you're combining both the context and some template metadata in one `@dataclass`. These are two separate concerns.

If you want to do it this way then use something like a facade pattern to isolate the templates entirely e.g.,

```
class MetadataTemplate(Template):
   @dataclass
   class Context(Template.Context):
      # The data that be fed into string .format
      benchmark_name: str

class Template:
    @dataclass
    class Context:
        pass . # Just a placeholder parent class for contexts 
    def __init__(self, *, context: Context, dest_file_name: str, template_file_name: str):
        pass

     def execute(self, *, force: Optional[bool] = False):
         # write the template here
         # can be overridden by subclasses
         # Don't actually do this, but equivalent: open(self.__dest_file_name).write(self.__template_file_name.read().format(self.__context))
```

Reply:
Are you saying to just write the template in the class? Presumably that would change how I defined my create_file_template function? 

And what do you mean by "don't actually do this?" Do you mean use different functions or eliminate the code completely?

Reply:
No, you can still have template files. But all of the logic for reading them, formatting the template with the context, and writing out the new file should be here, in Template or its subclasses.

"Don't actually do this": use with blocks instead of the single line.